src/pages/communication-page/communication.ts

```diff
@@ -153,55 +156,131 @@ export class CommunicationPage extends BasePageComponent {
     const {
       candidateProvidedEmail$,
       communicationEmail$,
+      communicationType$,
     } = this.pageState;
 
     const merged$ = merge(
       candidateProvidedEmail$.pipe(map(value => this.candidateProvidedEmail = value)),
       communicationEmail$.pipe(map(value => this.communicationEmail = value)),
+      communicationType$.pipe(map(value => this.communicationType = value)),
     );
     this.subscription = merged$.subscribe();
+
+    this.initialiseDefaultSelections();
+    this.restoreRadiosFromState();
+    this.restoreRadioValidators();
   }
 
   ngOnDestroy(): void {
     this.subscription.unsubscribe();
   }
 
   onSubmit() {
-    this.navController.push('WaitingRoomPage');
+    Object.keys(this.form.controls).forEach(controlName => this.form.controls[controlName].markAsDirty());
+    if (this.form.valid) {
+      this.navController.push('WaitingRoomPage');
+    }
   }
 
-  dispatchCandidateChoseProvidedEmail(emailType: string) {
-    this.setCommunicationType(emailType);
+  dispatchCandidateChoseProvidedEmail(communicationType: string) {
+    this.setCommunicationType(communicationType, this.providedEmail);
```

Comment:
dispatchCandidateChoseProvidedEmail is passed a string - but it is only ever 'Email', so it feels like the parameter is unnecessary.